### ISSUE TITLE
Feature Request: Context aware clicking

### ISSUE THREAD

**Issue by aaronrolls** [godbout/Wooshy.docs#34]
While I now use Wooshy for most things, I still will default to Homerow if I encounter a checkbox or a list of buttons that have the same name. Tabbing isn't fun. I encounter this multiple times a day because of my work so it may not reflect everybody's experience.

Most checkboxes, in apps and web, reside in a group with other things, sometimes text, sometimes links. What I would like is for Wooshy to be aware of which groups the last selection was in, then if you type `tick` or `check` for the checkbox, Wooshy will select the closest item to the previous selection (or the item in the closest parent group). You would still be able to tab and shift tab to other things Wooshy has detected of course, but I feel like something like this would be very handy.

This would also work well with the new `Command` `Shift` `Return` feature to move the mouse.

**Comment by godbout** [comment]
> While I now use Wooshy for most things, I still will default to Homerow if I encounter a checkbox or a list of buttons that have the same name. Tabbing isn't fun. I encounter this multiple times a day because of my work so it may not reflect everybody's experience.

just FYI, you can also use `up/down`, but if you're a kV user, you can use `jk`, `CTRL-F`, `CTRL-U` etc, and if you put Ws into the Key Mapping Family (currently), you can use `G/gg` to reach first and last targets. (https://github.com/godbout/Wooshy.docs#navigate)

> What I would like is for Wooshy to be aware of which groups the last selection was in, then if you type `tick` or `check` (or what ever each app chooses to name them) for the checkbox, Wooshy will select the closest item to the previous selection (or the item in the closest shared parent group).

that seems like something very specific to your workflow. you mean between launches Ws would remember what was the last group/target targeted?

**Comment by godbout** [comment]
>  I encounter a checkbox or a list of buttons that have the same name.

now that i'm thinking about it. do you mean a checkbox and a list have the same name (happens frequently), or do you mean a whole list of checkboxes and/or links have the same name? if it's the latter, in which specific case does this occur?

if it's the former, that's what the _narrowers_ are for. e.g.:

https://user-images.githubusercontent.com/121373/181668895-dc8f7923-1f2d-408f-9096-635ac45d4107.mp4

explained [here](https://github.com/godbout/Wooshy.docs#search) and [here](https://youtu.be/wzcsXKZ9ZDc?t=88)

that being said Homerow is gonna be way better in cases with lots of collisions, and stuff without AX info like in Things 3.

(btw knowing your skills i assume you've already seen the above but hey, still worth a shot.)

**Comment by aaronrolls** [comment]
> you mean between launches Ws would remember what was the last group/target targeted?

Yes, and it would highlight the most relevant/closest selection first.

Let's say Wooshy remembers the last two parent groups of its last target. If what you type applies to something in one of those parent groups, Wooshy highlights that one first. If it can't find anything belonging to those parent groups it acts as normal.

https://user-images.githubusercontent.com/47949221/181671442-af372913-4ca4-4c48-b223-b42068eb52d5.mov

This is just one example, but as you can see I have a lot of possible boxes to tick. I would like to be able to type `15` move the curser then type `tick` or whatever is appropriate and have the closest box selected first.

I tried using narrower's but they don't seem to work in these situations.

**Comment by godbout** [comment]
> I encounter this multiple times a day because of my work

just an idea. have you considered, like, not working?

> Yes, and it would highlight the most relevant/closest selection first.
> 
> Let's say Wooshy remembers the last two parent groups of its last target. If what you type applies to something in one of those parent groups, Wooshy highlights that one first. If it can't find anything belonging to those parent groups it acts as normal.

yeah ok. i need to digest this. currently the whole app is stateless. once The Input is gone, everything get cleaned. so that would require a major change. not for today. i'll need to mull it over and digest for a little while.

**Comment by godbout** [comment]
> Wooshy.Demo.mov

so i've spent ages building a way for users to custom The Input and you're using the default one.

![IMG_2480](https://user-images.githubusercontent.com/121373/181674928-f75774b4-a6d1-4f1f-bb33-a5f53da6d9a1.jpeg)

what AX info do you have in those tickboxes? nothing right?
(also is that a webpage? the checkboxes out of view are a little meh.)

**Comment by aaronrolls** [comment]
It is a web page yes. Most of my work involves using web apps, unfortunately.

![CleanShot2022-07-29at15 18 49](https://user-images.githubusercontent.com/47949221/181675804-5d6ab2b0-ead2-4bbb-b593-de5d06bb8ff2.png)

**Comment by godbout** [comment]
yeah, that's exactly the type of things where Homerow is gonna be better. they should add the date in the `Label/Value`. although you still have different rows with the same value.

**Comment by godbout** [comment]
@aaronrolls btw you're on Monterey 12.5? seems they've renamed `checkboxes` by `tickboxes` (Apple).
